owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/security/PassCodeViewModel.kt

```diff
@@ -51,11 +56,14 @@ class PassCodeViewModel(
         return isValid
     }
 
-    private fun loadPinFromOldFormatIfPossible(): String {
-        var pinString = ""
-        for (i in 1..4)
-            pinString += preferencesProvider.getString(PassCodeActivity.PREFERENCE_PASSCODE_D + i, null)
+    fun getNumberOfPassCodeDigits() = contextProvider.getInt(R.integer.passcode_digits)
 
-        return pinString
+    private fun loadPinFromOldFormatIfPossible(): String? {
+        var pinString = ""
+        for (i in 1..4) {
+            val pinChar = preferencesProvider.getString(PassCodeActivity.PREFERENCE_PASSCODE_D + i, null)
+            pinChar?.let { pinString += pinChar }
+        }
+        return if (pinString == "") null else pinString
```

Comment:
`isEmpty()` is our friend 👍 
```suggestion
        return if (pinString.isEmpty()) null else pinString
```